### ISSUE TITLE
Multiselect not populating selected values on re-load

### ISSUE THREAD

**Issue by tommym9** [LaravelCollective/html#479]
Laravel Version: 5.4.28
PHP Version: 7.1.4-1+deb.sury.org~xenial+1
Database Driver & Version: MySQL Community Edition - Standard (GPL) 5.7.19
Description:
When creating a multi-select field with Form::select() the submitted values aren't selected by default on reload.

This isn't the case for single select elements.

Steps To Reproduce:
Create a route that will accept GET parameters to filter a set of results
Add a form {!! Form::open(array('ROUTE_NAME', 'method' => 'get')) !!}
Add a single select field {!! Form::select('status', [null => '-- Filter by Status --'] + $statuses, null, ['placeholder' => 'Filter by Status']) !!}
Add a multi-select field {!! Form::select('years[]', [null => '-- Filter by Year(s) --'] + $report_years, null, ['placeholder' => 'Filter by Year(s)', 'multiple' => true]) !!}
Add a submit button {!! Form::submit('Go', ['class' => 'btn btn--small']) !!}
Expected behaviour:
The submitted options on both select fields are selected
Actual behaviour:
The submitted value for the single select is selected
None of the multi-select values are selected
I note that I can pass a third parameter to the select element

{!! Form::select('years[]', [null => '-- Filter by Year(s) --'] + $report_years, [$value1, $value2], ['placeholder' => 'Filter by Year(s)', 'multiple' => true]) !!}

And this will set the default value, however this is ignored on form submit.

Cheers

Tom

**Comment by nanocell** [comment]
This sounds like the same issue (bug) that i've encountered with multiselect elements. I'm going to hazard a guess and assume the keys in your $report_years array are integer values?

In my case the keys in that array were integer values and those "numeric" values are, upon form submission, received as strings. The culprits, I believe, lies in this line: [https://github.com/LaravelCollective/html/blob/38f2c6ff0d3dfc497175a5cff5e24b3937f6e2c9/src/FormBuilder.php#L786](url)

For some reason the in_array() function calls in those lines have been set to "strict" checking which conflicts with PHP's automatic casting of "numeric strings" to integer values when used as array keys. The fix for this issue would be to *not* enforce strict checking within the in_array() calls.

To summarize instead of doing this:
```
if (is_array($selected)) {
            return in_array($value, $selected, true) || in_array((string) $value, $selected, true) ? 'selected' : null;
```
I believe the code should be doing this:
```
if (is_array($selected)) {
            return in_array($value, $selected) || in_array((string) $value, $selected) ? 'selected' : null;
```

HTH.

**Comment by decadence** [comment]
Is this the same issue with strict comparision?
https://github.com/LaravelCollective/html/issues/500
